src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java

```diff
@@ -130,13 +130,13 @@ public BlocklyPanel(MCreator mcreator) {
 
 					// @formatter:off
 					webEngine.executeScript("var MCR_BLCKLY_PREF = { "
-							+ "'comments' : " + PreferencesManager.PREFERENCES.blockly.enableComments + ","
-							+ "'renderer' : '" + PreferencesManager.PREFERENCES.blockly.blockRenderer.toLowerCase(Locale.ENGLISH) + "',"
-							+ "'collapse' : " + PreferencesManager.PREFERENCES.blockly.enableCollapse + ","
-							+ "'trashcan' : " + PreferencesManager.PREFERENCES.blockly.enableTrashcan + ","
-							+ "'maxScale' : " + PreferencesManager.PREFERENCES.blockly.maxScale / 100.0 + ","
-							+ "'minScale' : " + PreferencesManager.PREFERENCES.blockly.minScale / 100.0 + ","
-							+ "'scaleSpeed' : " + PreferencesManager.PREFERENCES.blockly.scaleSpeed / 100.0 + ","
+							+ "'comments' : " + PreferencesManager.PREFERENCES.enableComments.get() + ","

```

Comment:
Now that you collapsed sub-categories, enableComments is too broad of a naming, no one can know this is for blockly editor. I think we should keep the hierarchy, or alternatively prefix all preferences to eg. blocklyEnableComments, etc.

Reply:
I put them in "categories" when declaring and initializing them, but I agree in the code, we can't really know directly.

Reply:
I pushed this commit https://github.com/MCreator/MCreator/pull/3072/commits/5d2f24dfaeff4936aef86fd4548b58dd4c5b0e7a to test something with the UI preference entries. Is it good and what do you want or would you change/improve it?
(Adding the type before each pref entry (e.g. `uiLanguage`) would require changing their name for all of them in the converter like I did here https://github.com/MCreator/MCreator/pull/3072/files#diff-8eb821d26c03f3b9dc25ef41f660d92830ebe3870a222ea4350fe7e54c76d07cR146)

Reply:
This will still prevent two entries of eg. maxScale, which is not ok. You should make it work in nested structure like it was before

Reply:
Changing everything to then being said to re-change everything once again is a waste of time. As I said, it was just a test with UI entries. So, I repeat my question, is the way I tested with UI entries **ONLY** correct and better for you? Don't check other types, I don't want feedback for them. I just want feedback about the UI category.

Reply:
You misunderstood me. The comment was for your linked commit suggestion.

I meant for UI, you just put them in a subclass, but in the properties file, they will still be maxScale only, not in subcategory, right?

Reply:
> You misunderstood me. The comment was for your linked commit suggestion.
> 
> I meant for UI, you just put them in a subclass, but in the properties file, they will still be maxScale only, not in subcategory, right?

Because changing it will require changing how it's made, so I want to have the final form before taking the time of doing this (big) change.

Reply:
In the code it is ok, but the generated properties file should also give this key either in JSON sub-object, or prefixed

Reply:
> In the code it is ok, external API is ok, but the generated properties file should also give this key either in JSON sub-object, or prefixed

I know. I didn't touch to it (and this is why it's totally broken in the current state)

Reply:
![image](https://user-images.githubusercontent.com/16374228/216685662-6d10c537-a801-4382-9a6f-2caeacc1088a.png)

This part is ok to me

Reply:
And is it ok too inside the `Preferences` class?

Reply:
Yes. Except maybe you could call register inside UI class itself, so you register files while defining them, would also visually split things a bit more. Section key String could also be moved inside this class (UI)

Reply:
I wasn't sure what would have been the best solution, so I kept the thing minimal.

Reply:
It's completed. I moved all entries to other classes, so we can have the category before the entry itself. However, as there are a lot of preferences (and we will have more in the future), I decided to make the classes new classes instead of subclasses (except for bedrock, as there's one entry currently). It makes things a bit cleaner.

Concerning the preference file, everything has been moved to a single file as requested. The file follows this format:
```
{
  "mcreator": {
    "ui": [
       *entries*
    ],
    ...
  },
  "plugin1": {...}
  ...
}
```